### PR TITLE
option: fix set sessionTimeout invalid

### PR DESCRIPTION
**Pull request by ddddddcf**
maybe it's optional

### CODE REVIEW

**Comment by mga-chka** [comment]
Hi,can you give more details about the bug you're trying to solve and how to produce it?

**Comment by mga-chka** [comment]
Moreover, could you add some unit tests in utils_test so that this part of the code is now covered?

**Comment by ddddddcf** [comment]
when I set the param "sessionTimeout“,it isn't used

**Comment by mga-chka** [comment]
ok, can you add a few unit tests in the files utils_test.go ?

File: utils_test.go [hunk 1]
```diff
@@ -279,6 +279,25 @@ func TestDecompression(t *testing.T) {
...
+	req, err := http.NewRequest(http.MethodPost, "http://127.0.0.1:9090", nil)
+	if err != nil {
+		panic(err)
+	}
+	params := make(url.Values)
+	params.Add("query", "SELECT 1")
+	params.Set("session_timeout", "888")
+	req.URL.RawQuery = params.Encode()
+	trueSessionTimeout := getSessionTimeout(req)
+	t.Logf("user set session_timeout true , return is %d",  trueSessionTimeout)
```
Comment (mga-chka) [inline utils_test.go]: if you only log, the test will pass even if the logic doesn't work inside getSessionTimeout().
So, if someone breaks this part, the automatic CI won't see it.

Can you rewrite the logic (for the 2 scenarios in your test) using a check like this one?
	sessionTimeout != [expected value] {
		t.Fatalf("[blabla] got: %q; expected %q", sessionTimeout, [expected value] )
	}

File: utils_test.go [hunk 2]
```diff
@@ -279,6 +279,35 @@ func TestDecompression(t *testing.T) {
...
+	req.URL.RawQuery = params.Encode()
+	if getSessionTimeout(req) != 60 {
+		t.Fatalf("user set session_timeout %q; get %q , expected %q", secondSessionTimeout, getSessionTimeout(req), 60)
+	}
+	params.Set("session_timeout", thirdSessionTimeout)
+	req.URL.RawQuery = params.Encode()
+	if getSessionTimeout(req) != 60 {
+		t.Fatalf("user set session_timeout %q; get %q , expected %q", thirdSessionTimeout, getSessionTimeout(req), 60)
+	}
+}
```
Comment (mga-chka) [inline utils_test.go:314]: Can you add a last test when there is no query param session_timeout (and the result should be 60)?
Then I'll merge your code.

**Review by mga-chka — APPROVED** [review]

**Pull request merged**